lib/internal/test_runner/snapshot.js

```diff
@@ -0,0 +1,242 @@
+'use strict';
+const {
+  ArrayPrototypeJoin,
+  ArrayPrototypeMap,
+  ArrayPrototypeSort,
+  JSONStringify,
+  ObjectKeys,
+  SafeMap,
+  String,
+  StringPrototypeReplaceAll,
+} = primordials;
+const {
+  codes: {
+    ERR_INVALID_STATE,
+  },
+} = require('internal/errors');
+const { emitExperimentalWarning, kEmptyObject } = require('internal/util');
+let debug = require('internal/util/debuglog').debuglog('test_runner', (fn) => {
+  debug = fn;
+});
+const {
+  validateArray,
+  validateFunction,
+  validateObject,
+} = require('internal/validators');
+const { strictEqual } = require('assert');
+const { mkdirSync, readFileSync, writeFileSync } = require('fs');
+const { dirname } = require('path');
+const { createContext, runInContext } = require('vm');
+const kExperimentalWarning = 'Snapshot testing';
+const kMissingSnapshotTip = 'Missing snapshots can be generated by rerunning ' +
+  'the command with the --test-update-snapshots flag.';
+const defaultSerializers = [
+  (value) => { return JSONStringify(value, null, 2); },
+];
+
+function defaultResolveSnapshotPath(testPath) {
+  if (typeof testPath !== 'string') {
+    return testPath;
+  }
+
+  return `${testPath}.snapshot`;
+}
+
+let resolveSnapshotPathFn = defaultResolveSnapshotPath;
+let serializerFns = defaultSerializers;
+
+function setResolveSnapshotPath(fn) {
+  emitExperimentalWarning(kExperimentalWarning);
+  validateFunction(fn, 'fn');
+  resolveSnapshotPathFn = fn;
+}
+
+function setDefaultSnapshotSerializers(serializers) {
+  emitExperimentalWarning(kExperimentalWarning);
+  validateFunctionArray(serializers, 'serializers');
+  serializerFns = ArrayPrototypeMap(serializers, (fn) => {
```

Comment:
why the mapping with identity function? binding the function? or am I missing anything?

Reply:
or if we want to clone the array, we can `ArrayPrototypeSlice(serializers, 0)`

Reply:
Yes, it's just to clone the array. I can update to use slice.